### PR TITLE
[CELEBORN-1248] Improve flusher's robustness

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
Improve flusher's robustness.

### Why are the changes needed?
Flusher's thread might be terminated due to uncaught exceptions.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
GA and cluaster.

### CODE REVIEW

**Review by SteNicholas — APPROVED** [review]

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/Flusher.scala [hunk 1]
```diff
@@ -64,27 +64,29 @@ abstract private[worker] class Flusher(
...
-                  case e: IOException =>
-                    task.notifier.setException(e)
-                    processIOException(e, DiskStatus.READ_OR_WRITE_FAILURE)
                 }
                 lastBeginFlushTime.set(index, -1)
               }
               returnBuffer(task.buffer)
               task.notifier.numPendingFlushes.decrementAndGet()
+            } catch {
+              case e: Throwable =>
```
Comment (pan3793) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/Flusher.scala:84]: moving the exception-catching block silently causes behavior change: line 84-87 skipped to execute when IOE throws now.

**Comment by FMX** [comment]
Updated. How about now? @pan3793

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/Flusher.scala [hunk 2]
```diff
@@ -76,14 +76,15 @@ abstract private[worker] class Flusher(
...
                   case e: IOException =>
                     task.notifier.setException(e)
                     processIOException(e, DiskStatus.READ_OR_WRITE_FAILURE)
+                  case t =>
+                    logWarning(s"Flusher-${this}-thread-${index} encounter exception.", t)
                 }
                 lastBeginFlushTime.set(index, -1)
               }
-              returnBuffer(task.buffer)
+              Utils.tryLogNonFatalError(returnBuffer(task.buffer))
```
Comment (pan3793) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/Flusher.scala:87]: in which cases, `returnBuffer` may throw exceptions? I saw other two places invoking it without handle any exceptions

Comment (FMX) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/Flusher.scala:90]: If a netty's buffer reference count is incorrect. This method will throw illegalreferencecountexception.

Comment (pan3793) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/Flusher.scala:90]: please add comment here

**Review by pan3793 — APPROVED** [review]
LGTM

**Review by AngersZhuuuu — APPROVED** [review]
LGTM

**Pull request closed without merging**